lib/Elastica/Type.php

```diff
@@ -299,6 +299,22 @@ public function setMapping($mapping)
     }
 
     /**
+     * Sets value type mapping for this type with timestamp .
+     *
+     * @param \Elastica\Type\Mapping|array $mapping Elastica\Type\MappingType object or property array with all mappings
+     *
+     * @return \Elastica\Response
+     */
+    public function setMappingWithTimeStamp($mapping,$timestamp)
```

Comment:
Instead of introducing a new function, I would suggest to add a param to setMapping